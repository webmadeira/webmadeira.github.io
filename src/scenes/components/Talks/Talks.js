import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import ResponsiveSquare from '../../../components/ResponsiveSquare/ResponsiveSquare'
import { H2 } from '../../../components/typography'

const TalksContainer = styled.div`
  display: flex;
  margin: 60px 0;
`

const NumTalks = styled.div`
  ${props => props.theme.mixins.center()}
  width: 100%;
  height: 100%;
  background-color: ${props => props.theme.color('secondary', 'light')};
  color: ${props => props.theme.color()};
`

const Total = styled.div`
  font-size: 60px;
  font-family: ${props => props.theme.fontFamily('secondary')}
`

const TalksFormatsContainer = styled.div`
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  margin-left: 20px;
`

const TalksFormat = styled.div`
  color: ${props => props.theme.color('neutral', 'bright')};
  border: solid 1px ${props => props.theme.color('secondary', 'light')};
  margin-bottom: 15px;
  flex-grow: 1;
  display: flex;
  align-items: center;
  justify-content: center;

  &:last-child {
    margin-bottom: 0;
  }
`

const Talks = ({ numTalks }) => (
  <TalksContainer>
    <ResponsiveSquare width="20%" minWidth="130px">
      <NumTalks>
        <Total>{numTalks}</Total>
        <H2>Talks</H2>
      </NumTalks>
    </ResponsiveSquare>
    <TalksFormatsContainer>
      <TalksFormat>{'5min { "Express talks" }'}</TalksFormat>
      <TalksFormat>{'20min { "Standard talks" }'}</TalksFormat>
    </TalksFormatsContainer>
  </TalksContainer>
)

Talks.propTypes = {
  numTalks: PropTypes.number.isRequired,
}

export default Talks
