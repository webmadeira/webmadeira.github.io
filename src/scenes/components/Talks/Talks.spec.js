import React from 'react'
import { shallow } from 'enzyme'
import Talks from './Talks'

describe('Talks component', () => {
  it('must render properly', () => {
    const component = shallow(<Talks numTalks={5} />)

    expect(component).toMatchSnapshot()
  })
})
