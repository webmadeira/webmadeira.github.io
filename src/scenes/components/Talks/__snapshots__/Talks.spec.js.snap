// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Talks component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Talks
    numTalks={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ResponsiveSquare
          minWidth="130px"
          style={Object {}}
          width="20%"
        >
          <styled.div>
            <styled.div>
              5
            </styled.div>
            <styled.h2>
              Talks
            </styled.h2>
          </styled.div>
        </ResponsiveSquare>,
        <styled.div>
          <styled.div>
            5min { "Express talks" }
          </styled.div>
          <styled.div>
            20min { "Standard talks" }
          </styled.div>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <styled.div>
            <styled.div>
              5
            </styled.div>
            <styled.h2>
              Talks
            </styled.h2>
          </styled.div>,
          "minWidth": "130px",
          "style": Object {},
          "width": "20%",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                5
              </styled.div>,
              <styled.h2>
                Talks
              </styled.h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 5,
              },
              "ref": null,
              "rendered": 5,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Talks",
              },
              "ref": null,
              "rendered": "Talks",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              5min { "Express talks" }
            </styled.div>,
            <styled.div>
              20min { "Standard talks" }
            </styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "5min { \\"Express talks\\" }",
            },
            "ref": null,
            "rendered": "5min { \\"Express talks\\" }",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "20min { \\"Standard talks\\" }",
            },
            "ref": null,
            "rendered": "20min { \\"Standard talks\\" }",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ResponsiveSquare
            minWidth="130px"
            style={Object {}}
            width="20%"
          >
            <styled.div>
              <styled.div>
                5
              </styled.div>
              <styled.h2>
                Talks
              </styled.h2>
            </styled.div>
          </ResponsiveSquare>,
          <styled.div>
            <styled.div>
              5min { "Express talks" }
            </styled.div>
            <styled.div>
              20min { "Standard talks" }
            </styled.div>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <styled.div>
              <styled.div>
                5
              </styled.div>
              <styled.h2>
                Talks
              </styled.h2>
            </styled.div>,
            "minWidth": "130px",
            "style": Object {},
            "width": "20%",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.div>
                  5
                </styled.div>,
                <styled.h2>
                  Talks
                </styled.h2>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": 5,
                },
                "ref": null,
                "rendered": 5,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Talks",
                },
                "ref": null,
                "rendered": "Talks",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                5min { "Express talks" }
              </styled.div>,
              <styled.div>
                20min { "Standard talks" }
              </styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "5min { \\"Express talks\\" }",
              },
              "ref": null,
              "rendered": "5min { \\"Express talks\\" }",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "20min { \\"Standard talks\\" }",
              },
              "ref": null,
              "rendered": "20min { \\"Standard talks\\" }",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
